File: 929-Unique-Email-Addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        emailsSet = set()

        for email in emails: 
            splitted = email.split('@')
            local = splitted[0].split('+')[0].replace('.','')
            domain = splitted[1]
            emailsSet.add((local, domain))

        return len(emailsSet)

File: neetcode/929-Unique-Email-Addresses.ts
function numUniqueEmails(emails: string[]): number {
  const unique: Set<string> = new Set();

  for (let email of emails) {
    // split the email into local and domain
    let [local, domain] = email.split('@')

    // remove all dots from the local and remove what's next the +
    local = local.replaceAll('.', '').split('+')[0]

    // add the correct formatted email to the unique set
    unique.add(local + '@' + domain)
  }

  return unique.size
};